src/compile/Component.ts

```diff
@@ -118,16 +118,15 @@ export default class Component {
 		this.componentOptions = process_component_options(this, this.ast.html.children);
 		this.namespace = namespaces[this.componentOptions.namespace] || this.componentOptions.namespace;
 
-		if (compileOptions.customElement === true && !this.componentOptions.tag) {
-			throw new Error(`No tag name specified`); // TODO better error
+		if (compileOptions.customElement) {
+			this.tag = this.componentOptions.tag || compileOptions.tag;
+			if (!this.tag) {
+				throw new Error(`Cannot compile to a custom element without specifying a tag name via options.tag or <svelte:options>`);
```

Comment:
This error message needs updating now

Reply:
Actually, this is also still checking for `compileOptions.tag`, which won't exist anymore

Reply:
Bluh, ignore both of these comments, oops sorry